cargo-shuttle/src/provisioner_server.rs

```diff
@@ -41,7 +41,7 @@ impl LocalProvisioner {
         // This only constructs the client and does not try to connect.
         // If the socket is not found, a "no such file" error will happen on the first request to Docker.
         Ok(Self {
-            docker: Docker::connect_with_local_defaults()?,
+            docker: Docker::connect_with_defaults()?,
```

Comment:
@fussybeaver Is the comment above this line still true for the new version? CI is failing for tests that shouldn't try to connect to Docker.

Reply:
I believe `connect_with_defaults` is a new method that tries to infer from environment variables whether you are connecting to a socket or https or a named pipe - IIRC it was added for better compatibility with testcontainers and co. I would stick to the old method if CI is failing, because we may not be setting those env vars appropriately.